internal/controller/messaging/subscriber.go

```diff
@@ -0,0 +1,178 @@
+			select {
+			case <-ctx.Done():
+				lc.Infof("Exiting waiting for MessageBus '%s' topic messages", requestTopic)
+				return
+			case err = <-messageErrors:
+				lc.Error(err.Error())
+			case msgEnvelope := <-messages:
+				lc.Debugf("Command request received on message queue. Topic: %s, Correlation-id: %s ", requestTopic, msgEnvelope.CorrelationID)
+
+				// expected command request topic scheme: #/<service-name>/<device-name>/<command-name>/<method>
+				topicLevels := strings.Split(msgEnvelope.ReceivedTopic, "/")
+				length := len(topicLevels)
+				if length < 4 {
+					lc.Error("Failed to parse and construct response topic scheme, expected request topic scheme: '#/<service-name>/<device-name>/<command-name>/<method>'")
+					continue
+				}
+
+				// expected command response topic scheme: #/<service-name>/<device-name>/<command-name>/<method>
+				serviceName := topicLevels[length-4]
+				deviceName := topicLevels[length-3]
+				commandName := topicLevels[length-2]
+				method := topicLevels[length-1]
+				responseTopic := strings.Join([]string{responseTopicPrefix, serviceName, deviceName, commandName, method}, "/")
+
+				var responseEnvelope types.MessageEnvelope
+				if strings.EqualFold(method, "get") {
+					var rawQuery string
+					for k, v := range msgEnvelope.QueryParams {
+						if rawQuery != "" {
+							rawQuery += "&"
+						}
+						rawQuery += fmt.Sprintf("%s=%s", k, v)
+					}
+
+					ctx = context.WithValue(ctx, correlationHeaderKey, msgEnvelope.CorrelationID)
+					events, edgexErr := application.GetCommand(ctx, deviceName, commandName, rawQuery, dic)
+					if edgexErr != nil {
+						responseEnvelope = types.NewMessageEnvelopeWithError(msgEnvelope.RequestID, edgexErr.Error())
+						err = messageBus.Publish(responseEnvelope, responseTopic)
+						if err != nil {
+							lc.Errorf("Failed to publish response: %s", err.Error())
+						}
+						continue
+					}
+					eventResponse := responses.NewEventResponse("", "", http.StatusOK, *events)
+					eventResponseBytes, encoding, err := eventResponse.Encode()
+					if err != nil {
+						responseEnvelope = types.NewMessageEnvelopeWithError(msgEnvelope.RequestID, err.Error())
+						err = messageBus.Publish(responseEnvelope, responseTopic)
+						if err != nil {
+							lc.Errorf("Failed to publish response: %s", err.Error())
+						}
+						continue
+					}
+
+					responseEnvelope, err = types.NewMessageEnvelopeForResponse(eventResponseBytes, msgEnvelope.RequestID, msgEnvelope.CorrelationID, encoding)
+					if err != nil {
+						responseEnvelope = types.NewMessageEnvelopeWithError(msgEnvelope.RequestID, err.Error())
+						err = messageBus.Publish(responseEnvelope, responseTopic)
+						if err != nil {
+							lc.Errorf("Failed to publish response: %s", err.Error())
+						}
+						continue
+					}
+
+					err = messageBus.Publish(responseEnvelope, responseTopic)
+					if err != nil {
+						lc.Errorf("Failed to publish response: %s", err.Error())
+					}
+				}
```

Comment:
Missing check for `PushEvent` query param is `yes` and then calling https://github.com/edgexfoundry/device-sdk-go/blob/e5efdee9dd9df15b49fe73e0060b5d971241be0d/internal/common/utils.go#L54